### DISCUSSION TITLE
What is the best upgrade path on a RHEL 7 server with NetBox version 2.10.8 to a RHEL 9 with NetBox version 4.0.7?

### DISCUSSION THREAD

**Discussion by ronthao3** [netbox-community/netbox#16874] [Help Wanted!]
Hi All,
I have a RHEL 7 server that is now EOL, and now we need to migrate to RHEL 9. What is the best upgrade path, and if anyone would be able to give me a step by step process I'd appreciate it. 
Thank you,
Ron

**Comment by candlerb** [comment]
1. Take a full backup of existing server.
2. Upgrade existing server to Netbox 2.11.12
3. Create a new server or VM with RHEL 9 and Netbox 3.7.8. Test it (it's OK to create items for testing)
4. Import the database and media/scripts from NB 2.11.12 into this - see [replicating Netbox](https://netboxlabs.com/docs/netbox/en/stable/administration/replicating-netbox/) in the docs - and run ./upgrade.sh. Check all is working correctly on this version.
5. Upgrade this server to Netbox 4.0.7

**Comment by ronthao3** [comment]
Thanks Brian! I will give this a try and keep you updated.On Jul 10, 2024, at 2:37 PM, Brian Candler ***@***.***> wrote:﻿

Take a full backup of existing server.
Upgrade existing server to Netbox 2.11.12
Create a new server or VM with RHEL 9 and Netbox 3.7.8. Test it (it's OK to create items for testing)
Import the database and media/scripts from NB 2.11.12 into this - see replicating Netbox in the docs - and run ./upgrade.sh. Check all is working correctly on this version.
Upgrade this server to Netbox 4.0.7

**Comment by ronthao3** [comment]
Hi Brian,Thanks a lot for your help! I was able to successfully get our NetBox data migrated to NetBox version 4.0.8.Thank you,Ronald ThaoOn Jul 16, 2024, at 11:58 AM, Ronald Thao ***@***.***> wrote:﻿Thanks Brian! I will give this a try and keep you updated.On Jul 10, 2024, at 2:37 PM, Brian Candler ***@***.***> wrote:﻿

Take a full backup of existing server.
Upgrade existing server to Netbox 2.11.12
Create a new server or VM with RHEL 9 and Netbox 3.7.8. Test it (it's OK to create items for testing)
Import the database and media/scripts from NB 2.11.12 into this - see replicating Netbox in the docs - and run ./upgrade.sh. Check all is working correctly on this version.
Upgrade this server to Netbox 4.0.7

**Comment by ronthao3** [comment]
Hi netbox-Community/Netbox support,

Is there any documentation for netbox software itself to integrate with
OKTA and assigning permissions at first login?

Thank you,
Ron Thao

**Reply by jnovinger** [reply]
Hey Ron, I would suggest starting with the [community Okta documentation](https://netboxlabs.com/docs/netbox/administration/authentication/okta/) for this.

The documentation doesn't mention automatically assigning permissions and I'm away from my laptop at the moment, so can't verify this bit.
